config/main.py

```diff
@@ -2591,15 +2611,16 @@ def _update_buffer_calculation_model(config_db, model):
     '--dry_run', type=click.STRING,
     help="Dry run, writes config to the given file"
 )
-def reload(ctx, no_dynamic_buffer, dry_run, json_data, ports):
+def reload(ctx, no_dynamic_buffer, no_delay, dry_run, json_data, ports):
     """Reload QoS configuration"""
     if ports:
         log.log_info("'qos reload --ports {}' executing...".format(ports))
         _qos_update_ports(ctx, ports, dry_run, json_data)
         return
 
     log.log_info("'qos reload' executing...")
-    _clear_qos()
+    if not dry_run:
+        _clear_qos(delay = no_delay)
```

Comment:
The usage is confusing. no_delay flag is set to True in load_minigraph, so the user expects that there will be no delay but the code eventually translates that no_delay to 'delay' and introduces a wait.
IMO, the call should be _clear_qos(no_delay=no_delay) and execute the wait only if '--no-delay' flag is not called in the command 'config qos reload'
The current usage of adding a worst case 30s time to load_minigraph which should be avoided

Reply:
@neethajohn  You are correct, that was suppose to be the other way around. Fixed the mistake, thank you.